python/cudf_polars/cudf_polars/utils/config.py

```diff
@@ -5,209 +5,306 @@
 
 from __future__ import annotations
 
-import copy
+import dataclasses
+import enum
 import json
-from typing import TYPE_CHECKING, Any
+from typing import TYPE_CHECKING, Literal
 
 if TYPE_CHECKING:
-    from collections.abc import MutableMapping, Sequence
-
     from typing_extensions import Self
 
+    import polars as pl
+
+
 __all__ = ["ConfigOptions"]
 
 
-class ConfigOptions:
+# TODO: Use enum.StrEnum when we drop Python 3.10
+
+
+class FallbackMode(str, enum.Enum):
```

Comment:
The `"fallback_mode"` config is currently used by the `"streaming"` executor only, and it is unrelated to CPU fallback. Rather, this tells the `"streaming"` engine how to handle fallback to single-partition execution (still on GPU).